crates/kubelet/src/pod/queue.rs

```diff
@@ -19,79 +30,166 @@ use crate::provider::Provider;
 /// modify will be handled, which is ok given that each event contains the whole pod object
 pub(crate) struct Queue<P> {
     provider: Arc<P>,
-    handlers: HashMap<String, Worker>,
-    error_sender: Sender<(KubePod, anyhow::Error)>,
+    handlers: HashMap<String, tokio::sync::mpsc::Sender<WatchEvent<KubePod>>>,
+    client: KubeClient,
 }
 
-struct Worker {
-    sender: watch::Sender<WatchEvent<KubePod>>,
-    _worker: JoinHandle<()>,
-}
+impl<P: 'static + Provider + Sync + Send> Queue<P> {
+    pub fn new(provider: Arc<P>, client: KubeClient) -> Self {
+        Queue {
+            provider,
+            handlers: HashMap::new(),
+            client,
+        }
+    }
 
-impl Worker {
-    fn create<P>(
+    async fn run_pod(
+        &self,
         initial_event: WatchEvent<KubePod>,
-        provider: Arc<P>,
-        mut error_sender: Sender<(KubePod, anyhow::Error)>,
-    ) -> Self
-    where
-        P: 'static + Provider + Sync + Send,
-    {
-        let (sender, mut receiver) = watch::channel(initial_event);
-        let worker = tokio::spawn(async move {
+    ) -> anyhow::Result<tokio::sync::mpsc::Sender<WatchEvent<KubePod>>> {
+        let (sender, mut receiver) = tokio::sync::mpsc::channel::<WatchEvent<KubePod>>(16);
```

Comment:
Do we need to handle multiple events or just the latest event since we'll reconcile it to the latest definition anyway?

Reply:
I agree that you only need to handle the latest event, but I also like that you don't have to initialize the channel with the first event, since we are using it immediately.

Reply:
Should we at least lower the buffer size then?

Reply:
We can, I felt like this was a reasonably small default.